native/cocos/renderer/pipeline/custom/RenderGraphTypes.h

```diff
@@ -728,10 +730,10 @@ struct RenderData {
     RenderData& operator=(RenderData&& rhs) = default;
     RenderData& operator=(RenderData const& rhs) = delete;
 
-    PmrUnorderedMap<uint32_t, ccstd::pmr::vector<uint8_t>>   constants;
-    PmrUnorderedMap<uint32_t, std::unique_ptr<gfx::Buffer>>  buffers;
-    PmrUnorderedMap<uint32_t, std::unique_ptr<gfx::Texture>> textures;
-    PmrUnorderedMap<uint32_t, std::unique_ptr<gfx::Sampler>> samplers;
+    PmrUnorderedMap<uint32_t, ccstd::pmr::vector<uint8_t>> constants;
+    PmrUnorderedMap<uint32_t, IntrusivePtr<gfx::Buffer>>   buffers;
+    PmrUnorderedMap<uint32_t, IntrusivePtr<gfx::Texture>>  textures;
+    PmrUnorderedMap<uint32_t, ObserverPtr<gfx::Sampler>>   samplers;
```

Comment:
Wha's ObserverPtr?

Reply:
ObserverPtr is wrapper of raw-pointer, it will be removed in next weeks.
Currently, pipeline generator cannot handle raw-pointer, ObserverPtr is a walk-around.

Reply:
Got it. Thanks.